src/main/java/teammates/logic/core/GoogleCloudLoggingService.java

```diff
@@ -242,13 +242,43 @@ private void createLogEntry(LogEntry entry) throws LogServiceException {
         if (s.endTime != null) {
             logFilters.add("timestamp<=\"" + s.endTime.toString() + "\"");
         }
-        if (s.severities != null) {
-            String severitiesFilter = s.severities.stream().collect(Collectors.joining(" OR "));
-            logFilters.add("severity=(" + severitiesFilter + ")");
+        if (s.severity != null) {
+            logFilters.add("severity=" + s.severity);
         }
-        if (s.minSeverity != null && s.severities == null) {
+        if (s.minSeverity != null && s.severity == null) {
             logFilters.add("severity>=" + s.minSeverity.toString());
         }
+        if (s.traceId != null) {
+            logFilters.add("trace=\"" + s.traceId + "\"");
+        }
+        if (s.actionClass != null) {
+            logFilters.add("jsonPayload.actionClass=\"" + s.actionClass + "\"");
+        }
+        if (s.userInfoParams != null) {
+            if (s.userInfoParams.getGoogleId() != null) {
+                logFilters.add("jsonPayload.userInfo.googleId=\"" + s.userInfoParams.getGoogleId() + "\"");
+            }
+            if (s.userInfoParams.getRegkey() != null) {
+                logFilters.add("jsonPayload.userInfo.regkey=\"" + s.userInfoParams.getRegkey() + "\"");
+            }
+            if (s.userInfoParams.getEmail() != null) {
+                logFilters.add("jsonPayload.userInfo.email=\"" + s.userInfoParams.getEmail() + "\"");
+            }
+        }
+        if (s.logEvent != null) {
+            logFilters.add("jsonPayload.event=\"" + s.logEvent + "\"");
+        }
+        if (s.sourceLocation != null && s.sourceLocation.getFile() != null) {
+            if (s.sourceLocation.getFunction() == null) {
+                logFilters.add("sourceLocation.file=\"" + s.sourceLocation.getFile() + "\"");
+            } else {
+                logFilters.add("sourceLocation.file=\"" + s.sourceLocation.getFile()
+                        + "\" AND sourceLocation.function=\"" + s.sourceLocation.getFunction() + "\"");
+            }
+        }
+        if (s.exceptionClass != null) {
+            logFilters.add("textPayload:\"" + s.exceptionClass + "\"");
```

Comment:
How valid is this assumption, e.g. an exception happening equals to the exception name being passed to the `textPayload`?

Reply:
I cannot find any documentation on this, but after examining many log entries, this is the case.

Reply:
That's a really risky conclusion, although it might be a lot of work to find another, more robust way.

Can you add a TODO here for that purpose?

Reply:
Ok. A TODO added.